charts/premium/authelia/templates/_configmap.tpl

```diff
@@ -245,9 +245,11 @@ data:
     {{- range $client := .Values.identity_providers.oidc.clients }}
         - id: {{ $client.id }}
           description: {{ $client.description | default $client.id }}
-          secret: {{ $client.secret | default (randAlphaNum 128) }}
           {{- if $client.public }}
           public: {{ $client.public }}
+          secret: ""
+          {{- else }}
+          secret: {{ $client.secret | default (randAlphaNum 128) }}
```

Comment:
```suggestion
          secret: {{ $client.secret  }}
```

Also needs a fail case

Reply:
if it's not set to public and secret key isn't set, it's supposed to be set (similar) to the format  created by running `authelia crypto hash generate pbkdf2 --variant sha512 --random --random.length 72 --random.charset rfc3986` though

Reply:
Yes, precisely.
Thats why a fail case should be added in case secret isn't set.
As it always should be set.

Reply:
random is NOT allowed here or *anywhere*
as it will randomize the key on EVERY edit of the helm deployment